src/net.cpp

```diff
@@ -703,12 +706,14 @@ CNetMessage V1TransportDeserializer::GetMessage(const CMessageHeader::MessageSta
     // We just received a message off the wire, harvest entropy from the time (and the message checksum)
     RandAddEvent(ReadLE32(hash.begin()));
 
-    msg.m_valid_checksum = (memcmp(hash.begin(), hdr.pchChecksum, CMessageHeader::CHECKSUM_SIZE) == 0);
-    if (!msg.m_valid_checksum) {
+    bool valid_checksum = (memcmp(hash.begin(), hdr.pchChecksum, CMessageHeader::CHECKSUM_SIZE) == 0);
+    if (!valid_checksum) {
         LogPrint(BCLog::NET, "CHECKSUM ERROR (%s, %u bytes), expected %s was %s\n",
                  SanitizeString(msg.m_command), msg.m_message_size,
                  HexStr(hash.begin(), hash.begin()+CMessageHeader::CHECKSUM_SIZE),
                  HexStr(hdr.pchChecksum, hdr.pchChecksum+CMessageHeader::CHECKSUM_SIZE));
+        Reset();
```

Comment:
Does this change behaviour in any way?

Reply:
Without this we fall into an infinite loop in one of the tests.  It ultimately stems from a combo of a few different members of V1TransportDeserializer interacting in such a way that we repeatedly fill and drain vRecv by the same amounts, without causing it to grow infinitely and without writing off of the end.